src/core/public/entry_point.ts

```diff
@@ -59,5 +66,21 @@ i18n
       setup.fatalErrors.add(i18nError);
     }
 
+    /**
+     * Register listeners for navigation changes and capture them as
+     * route-change transactions after Kibana app is bootstrapped
+     */
+    if (APM_ENABLED) {
+      coreSystem.application.currentAppId$.subscribe((appId) => {
```

Comment:
@elastic/kibana-platform I am accessing a private read only field here since I thought it would be better to have the logic in the entrypoint instead of putting it straight in `navigateToApp` where navigation changes. Do you think it would be fine to have it `public readonly` field till the platform exposes navigations as events to the consumers?

Reply:
Accessing the services instances directly should be avoided at all cost. Only the exposed contracts should be considered stable.

`currentAppId$` is available on the public start contract of the `application` service.

As it's already done for the `fatalError` service in `coreSystem.setup`, I would return the `application` service start contract from `coreSystem.start`.

`src/core/public/core_system.tsL296`

```ts
return {
  application,
};
```

`src/core/public/kbn_bootstrap.ts` (previously `src/core/public/entry_point.ts`, PR is outdated) L62

```ts
const start = await coreSystem.start();
if(start) {
    start.application.currentAppId$.subscribe(/* ... */);
}
```

Reply:
Nice, that would solve the RUM use-case. Thanks @pgayvallet for the pointers.